opentracing-spring-cloud-starter-jaeger-it/src/test/java/org/example/SpringWebApplicationTracesCollectedTest.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright 2017-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.example;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.EnvironmentTestUtils;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.LogMessageWaitStrategy;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(
+    classes = DemoSpringBootWebApplication.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ContextConfiguration(initializers = SpringWebApplicationTracesCollectedTest.Initializer.class)
+public class SpringWebApplicationTracesCollectedTest {
+
+
+  private static final int QUERY_PORT = 16686;
+  private static final int COLLECTOR_PORT = 14268;
+  private static final String SERVICE_NAME = "spring-boot-test";
+
+  @ClassRule
+  public static GenericContainer jaeger
+      = new GenericContainer("jaegertracing/all-in-one:1.3")
+      .withExposedPorts(COLLECTOR_PORT, QUERY_PORT)
+      //make sure we wait until the collector is ready
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*jaeger-query.*HTTP server.*\n"));
```

Comment:
There is a status endpoint which returns the correct state of the system. This string can change

Reply:
I didn't know that, thanks!

Reply:
I couldn't get the status endpoint to work for me so I left it as is for the time being....

Reply:
apologize, actually all-in-one does not expose it. I have created https://github.com/jaegertracing/jaeger/issues/775

Reply:
:+1: